plugins/destination/clickhouse/queries/types.go

```diff
@@ -157,9 +157,8 @@ func normalizeColumn(column schema.Column) schema.Column {
 	}
 
 	res.CreationOptions = column.CreationOptions
-	if !res.CreationOptions.NotNull {
-		res.CreationOptions.NotNull = column.CreationOptions.PrimaryKey
-	}
+	// We only support append mode so we don't care about primary key changes to the schema
+	res.CreationOptions.PrimaryKey = false
```

Comment:
Does SDK make all PKs be not null?
In our case it would be beneficial to have this.

Now you're changing the behaviour, so I most certainly would mark this as a breaking change (consider the sync 1 where the col was a PK and was created as non-nullable, and then this version syncs and has to drop not null) (so, essentially, drops the data).

Reply:
Great point! Re-added in https://github.com/cloudquery/cloudquery/pull/8255/commits/ba006d7c777bfc06f81a746522c449991393e481